runtime/observability/telemetry-provider-otel/jvm/test/aws/smithy/kotlin/runtime/telemetry/otel/AttributeTests.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.telemetry.otel
+
+import aws.smithy.kotlin.runtime.util.AttributeKey
+import aws.smithy.kotlin.runtime.util.attributesOf
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlin.test.assertNotNull
+import io.opentelemetry.api.common.AttributeKey as OtelAttributeKey
+
+class AttributeTests {
+    @Test
+    fun testToOtelAttributeKey() {
+        val strAttr = AttributeKey<String>("key")
+        val otelStrKey = strAttr.otelAttrKeyOrNull("foo")
+        assertNotNull(otelStrKey)
+        assertIs<OtelAttributeKey<String>>(otelStrKey)
+
+        val longAttr = AttributeKey<Long>("key")
+        val otelLongKey = longAttr.otelAttrKeyOrNull(0L)
+        assertNotNull(otelLongKey)
+        assertIs<OtelAttributeKey<Long>>(otelLongKey)
+
+        val boolAttr = AttributeKey<Boolean>("key")
+        val otelBoolKey = boolAttr.otelAttrKeyOrNull(true)
+        assertNotNull(otelBoolKey)
+        assertIs<OtelAttributeKey<Long>>(otelBoolKey)
```

Comment:
**Nit**: The generic type parameter checks are meaningless in these `assertIs` calls because of type erasure (as evidenced by the fact that the presumably-erroneous `assertIs<OtelAttributeKey<Long>>(otelBoolKey)` check passes). Without reflection, I think the best we can assert is `OtelAttributeKey<*>`.

Reply:
🤦